### PR TITLE
Add option to dynamically update the baseline of the RJRInit plugin

### PR DESCRIPTION
**Pull request by jtnord**
if the version of Jenkins used by `RealJenkinsRuleInit` has detached plugins in the version of Jenkins used for the test then this can unnecessarily pull in detached plugins.

In most cases this is harmless, but in some cases these plugins may be prevented from running under the test (e.g. testing FIPS and one of the detached plugins like `eddsa` is not FIPS complaint and blocks start-up).

The plugin itself has limited exposure to the Jenkins api (via `hudson.plugin.Plugin` and `RealJenkinsRule$Init2`).

Code running from the plugin tests inside `RealJenkinsRule` uses the `uberClassloader` and if the plugin was targeting a lower core (where the dependencies have been detached, they should still be included as this change only affects the version of the `RealJenkinsRuleInit` plugin.

this is a "future proof" (but potentially more risky) version of #779 (nb. the changes are not mutually exclusive)

consumed in https://github.com/jenkinsci/jenkins/pull/9353

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java [hunk 1]
```diff
@@ -199,6 +200,7 @@ public final class RealJenkinsRule implements TestRule {
 
     private boolean prepareHomeLazily;
     private boolean provisioned;
+    private boolean updateRealJenkinsRuleInitPluginBaseline;
```
[the remark this answers is not in the document]
Comment (timja) [reply src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:203]: is it worth just running this through bom with it enabled by default? this doesn't sound like something you want to have to try dig out when it should "just work".

[the remark this answers is not in the document]
Comment (jtnord) [reply src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:203]: https://github.com/jenkinsci/jenkins-test-harness/pull/781  I expect anything using a JNLP based agent and an updated baseline to fail...

[the remark this answers is not in the document]
Comment (jtnord) [reply src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:203]: https://github.com/jenkinsci/bom/pull/3284

File: src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java [hunk 2]
```diff
@@ -514,7 +526,16 @@ private void provision() throws Exception {
         }
         File plugins = new File(getHome(), "plugins");
         Files.createDirectories(plugins.toPath());
-        FileUtils.copyURLToFile(RealJenkinsRule.class.getResource("RealJenkinsRuleInit.jpi"), new File(plugins, "RealJenkinsRuleInit.jpi"));
+        if (updateRealJenkinsRuleInitPluginBaseline) {
+            try (JarFile jf = new JarFile(war);
+                    InputStream resourceAsStream = RealJenkinsRule.class.getResourceAsStream("RealJenkinsRuleInit.jpi");
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:531]: Actually with https://github.com/jenkinsci/jenkins-test-harness/pull/781#discussion_r1628435294 we could clean up some longstanding tech debt and get rid of the `.jpi` file in Git: just compile the one class from sources in this harness (include in `src/main/java/`) and pack the plugin on the fly.

**Comment by jtnord** [comment]
closing in favor of #782

**Pull request closed without merging**